### PR TITLE
Added Chainlink aggregator with tests

### PR DESCRIPTION
**Pull request by yashpatel5400**
PR for changes requested in https://gitcoin.co/issue/Agoric/agoric-sdk/3200/100026006

### CODE REVIEW

File: packages/zoe/tools/types.js [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * @typedef {Object} PriceDescription A description of a single quote
  * @property {Amount} amountIn The amount supplied to a trade
  * @property {Amount} amountOut The quoted result of trading `amountIn`
- * @property {TimerService} timer The service that gave the `timestamp`
+ * @property {ERef<TimerService>} timer The service that gave the `timestamp`
```
Comment (michaelfig) [inline packages/zoe/tools/types.js]: I suggest reverting this.  The PriceDescription needs to be comparable, and that means the `timer` should be `await`ed before putting it in the description.

File: packages/zoe/src/contracts/priceAggregatorChainlink.js [hunk 2]
```diff
@@ -0,0 +1,754 @@
...
+
+/**
+ * This contract aggregates price values from a set of oracleStatuses and provides a
+ * PriceAuthority for their median.
+ *
+ * @type {ContractStartFn}
+ */
+const start = async zcf => {
+  const {
+    timer: rawTimer,
```
Comment (michaelfig) [inline packages/zoe/src/contracts/priceAggregatorChainlink.js:28]: Sorry to do this, but I think we should pivot a little bit:

```suggestion
    timer: timerP,
```

File: packages/zoe/src/contracts/priceAggregatorChainlink.js [hunk 3]
```diff
@@ -0,0 +1,754 @@
...
+      const {
+        amountIn,
+        amountOut,
+        timestamp: theirTimestamp = timestamp,
+      } = quote;
+      AmountMath.coerce(amountIn, brandIn);
+      AmountMath.coerce(amountOut, brandOut);
+      if (theirTimestamp !== undefined) {
+        return authenticateQuote([
+          { amountIn, amountOut, timer, timestamp: theirTimestamp },
```
Comment (michaelfig) [inline packages/zoe/src/contracts/priceAggregatorChainlink.js]: This is how we use the non-ERef form of the timer as an identity:

```suggestion
          { amountIn, amountOut, timer: timerPresence, timestamp: theirTimestamp },
```

File: packages/zoe/src/contracts/priceAggregatorChainlink.js [hunk 4]
```diff
@@ -0,0 +1,754 @@
...
+    maxSubmissionValue,
+
+    unitAmountIn = AmountMath.make(1n, brandIn),
+  } = zcf.getTerms();
+
+  const unitIn = AmountMath.getValue(unitAmountIn, brandIn);
+
+  /** @type {ERef<TimerService>} */
+  const timer = rawTimer;
+
```
Comment (michaelfig) [inline packages/zoe/src/contracts/priceAggregatorChainlink.js:44]: I may have slightly misled you... we do actually need the non-ERef TimerService to put it in the quotes.  I suggest doing this:

```suggestion

  // Get the timer's identity.
  const timerPresence = await timer;

```

File: packages/zoe/src/contracts/priceAggregatorChainlink.js [hunk 5]
```diff
@@ -0,0 +1,754 @@
...
+      AmountMath.coerce(amountOut, brandOut);
+      if (theirTimestamp !== undefined) {
+        return authenticateQuote([
+          { amountIn, amountOut, timer, timestamp: theirTimestamp },
+        ]);
+      }
+      return E(timer)
+        .getCurrentTimestamp()
+        .then(now =>
+          authenticateQuote([{ amountIn, amountOut, timer, timestamp: now }]),
```
Comment (michaelfig) [inline packages/zoe/src/contracts/priceAggregatorChainlink.js]: Please do the same with `timer: timerPresence` here.

**Comment by yashpatel5400** [comment]
Addressed comments

**Review by michaelfig — APPROVED** [review]
Beautiful!

**Pull request merged**